File: lib/log.js
// Unified color logging utility

const c = {
  reset: "\x1b[0m",
  cyan: "\x1b[36m",
  green: "\x1b[32m",
  yellow: "\x1b[33m",
  red: "\x1b[31m",
  magenta: "\x1b[35m",
  gray: "\x1b[90m",
  bold: "\x1b[1m",
};

const log = {
  info: (msg, ...args) =>
    console.log(`${c.cyan}[INFO]${c.reset} ${msg}`, ...args),
  ok: (msg, ...args) =>
    console.log(`${c.green}[ OK ]${c.reset} ${msg}`, ...args),
  warn: (msg, ...args) =>
    console.warn(`${c.yellow}[WARN]${c.reset} ${msg}`, ...args),
  err: (msg, ...args) =>
    console.error(`${c.red}[ERR ]${c.reset} ${msg}`, ...args),
  req: (msg, ...args) =>
    console.log(`${c.magenta}[REQ ]${c.reset} ${msg}`, ...args),
  resp: (msg, ...args) =>
    console.log(`${c.green}[RESP]${c.reset} ${msg}`, ...args),
  skip: (msg, ...args) =>
    console.log(`${c.gray}[SKIP]${c.reset} ${msg}`, ...args),
  toks: (prompt, completion, total) => {
    const parts = [];
    if (prompt != null) parts.push(`in:${prompt}`);
    if (completion != null) parts.push(`out:${completion}`);
    if (total != null) parts.push(`total:${total}`);
    console.log(`${c.gray}[TOKS]${c.reset} ${parts.join(" ")}`);
  },
  header: (msg) =>
    console.log(`\n${c.bold}${c.cyan}=== ${msg} ===${c.reset}`),
};

export default log;

File: log.go
package main

import (
	"fmt"
	"os"
)

// Color codes for terminal output
const (
	cReset   = "\033[0m"
	cCyan    = "\033[36m"
	cGreen   = "\033[32m"
	cYellow  = "\033[33m"
	cRed     = "\033[31m"
	cMagenta = "\033[35m"
	cGray    = "\033[90m"
	cBold    = "\033[1m"
)

var quietMode bool

func logInfo(msg string, args ...interface{}) {
	fmt.Printf(cCyan+"[INFO]"+cReset+" "+msg+"\n", args...)
}

func logOk(msg string, args ...interface{}) {
	fmt.Printf(cGreen+"[ OK ]"+cReset+" "+msg+"\n", args...)
}

func logWarn(msg string, args ...interface{}) {
	fmt.Fprintf(os.Stderr, cYellow+"[WARN]"+cReset+" "+msg+"\n", args...)
}

func logErr(msg string, args ...interface{}) {
	fmt.Fprintf(os.Stderr, cRed+"[ERR ]"+cReset+" "+msg+"\n", args...)
}

func logReq(msg string, args ...interface{}) {
	if quietMode {
		return
	}
	fmt.Printf(cMagenta+"[REQ ]"+cReset+" "+msg+"\n", args...)
}

func logResp(msg string, args ...interface{}) {
	if quietMode {
		return
	}
	fmt.Printf(cGreen+"[RESP]"+cReset+" "+msg+"\n", args...)
}

func logSkip(msg string, args ...interface{}) {
	fmt.Printf(cGray+"[SKIP]"+cReset+" "+msg+"\n", args...)
}

func logToks(prompt, completion, total int) {
	parts := ""
	if prompt > 0 {
		parts += fmt.Sprintf("in:%d ", prompt)
	}
	if completion > 0 {
		parts += fmt.Sprintf("out:%d ", completion)
	}
	if total > 0 {
		parts += fmt.Sprintf("total:%d", total)
	}
	fmt.Printf(cGray+"[TOKS]"+cReset+" %s\n", parts)
}

func bold(s string) string {
	return cBold + s + cReset
}

func cyan(s string) string {
	return cCyan + s + cReset
}
